### ISSUE TITLE
Broken Pipe every 100-300 messages

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ntrees/javapns#10]
```
I'm running send notification in a tight loop to try and benchmark the 
performance/throughput.  I'm 
seeing a broken pipe exception fairly often.  This limits my throughput to 
around 100-300 
messages per second, with significant overhead in re-establishing the 
connection.  I've enabled 
TCP keepalive, but this doesn't seem to have effected performance at all.
```

Original issue reported on code.google.com by `str...@gmail.com` on 1 Dec 2009 at 4:38

**Comment by GoogleCodeExporter** [comment]
Original comment by `idbill.p...@gmail.com` on 27 Apr 2010 at 4:39
- Added labels: **Priority-Low**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Solved in 2.0.  The new javapns.notification.transmission.NotificationThread 
class provides the ability to limit and manage automatically the number of 
notifications per connection, to avoid this issue with Apple servers.
```

Original comment by `sype...@gmail.com` on 9 Sep 2011 at 7:03
- Changed state: **Fixed**
